compiler/compile/OMRCompilation.hpp

```diff
@@ -1262,6 +1275,11 @@ class OMR_EXTENSIBLE Compilation
 
    BitVectorPool _bitVectorPool; //MUST be declared after _trMemory
 
+   typedef TR::typed_allocator<std::pair<TR_OpaqueClassBlock* const, TR::TypeLayout *>, TR::Region &> LayoutAllocator;
+   typedef std::less<TR_OpaqueClassBlock*> LayoutComparator;
+   typedef std::map<TR_OpaqueClassBlock *, TR::TypeLayout *, LayoutComparator, LayoutAllocator> TypeLayoutMap;
```

Comment:
If we're going to return `const TR::TypeLayout*`, then the `const` has to be used everywhere. I believe this will fix the build failure:
```suggestion
   typedef std::map<TR_OpaqueClassBlock *, const TR::TypeLayout *, LayoutComparator, LayoutAllocator> TypeLayoutMap;
```